client/client.go

```diff
@@ -0,0 +1,142 @@
+package client
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"search/indexer"
+	"search/server"
+	"search/util"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// Client stores the necessary information for a client.
+type Client struct {
+	directory    string                      // Directory for the client where all the files are stored
+	server        *server.Server              // The server that this client is connected to
+	indexer       *indexer.SecureIndexBuilder // The indexer for the client
+	lookupTable   map[string]string           // A map from document ids to actual filenames
+	reverseLookup map[string]string           // A map from actual filenames to document ids
+}
+
+// CreateClient instantiates a client connected to Server `s` with a
+// `clientNum`.
```

Comment:
Might be worth mentioning here that the client instance can't be saved and reused after a switch to another client, since there's nothing to force the re-reading of the lookup table.  A new client must be constructed on every client switch.